jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java

```diff
@@ -0,0 +1,427 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.query.text;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.util.NodeFactoryExtra;
+import org.elasticsearch.action.admin.indices.exists.indices.IndicesExistsRequest;
+import org.elasticsearch.action.admin.indices.exists.indices.IndicesExistsResponse;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.update.UpdateRequest;
+import org.elasticsearch.action.update.UpdateResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.get.GetField;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.util.*;
+
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+
+/**
+ * Elastic Search Implementation of {@link TextIndex}
+ *
+ */
+public class TextIndexES implements TextIndex {
+
+    /**
+     * The definition of the Entity we are trying to Index
+     */
+    private final EntityDefinition docDef ;
+
+    /**
+     * Thread safe ElasticSearch Java Client to perform Index operations
+     */
+    private static Client client;
+
+    /**
+     * The name of the index. Defaults to 'test'
+     */
+    private final String INDEX_NAME;
+
+    static final String CLUSTER_NAME = "cluster.name";
+
+    static final String NUM_OF_SHARDS = "number_of_shards";
+
+    static final String NUM_OF_REPLICAS = "number_of_replicas";
+
+    private boolean isMultilingual ;
+
+    private static final Logger LOGGER      = LoggerFactory.getLogger(TextIndexES.class) ;
+
+    public TextIndexES(TextIndexConfig config, ESSettings esSettings) throws Exception{
+
+        this.INDEX_NAME = esSettings.getIndexName();
+        this.docDef = config.getEntDef();
+
+
+        this.isMultilingual = config.isMultilingualSupport();
```

Comment:
I wonder whether this class works properly when multilingual mode is disabled. Particularly for the `rdfs:label "Berlin"@de, "Berlin"@en` case where one of them is removed, does the index still understand that one "Berlin" value should be retained? I suggest that what is currently the multilingual mode should be the only possible mode.